tests/integrations/strawberry/test_strawberry_py3.py

```diff
@@ -0,0 +1,650 @@
+        )
+        return sync_app.test_client()
+
+    return create_app
+
+
+def test_async_execution_uses_async_extension(sentry_init):
+    sentry_init(integrations=[StrawberryIntegration(async_execution=True)])
+
+    with mock.patch(
+        "sentry_sdk.integrations.strawberry._get_installed_modules",
+        return_value={"flask": "2.3.3"},
+    ):
+        # actual installed modules should not matter, the explicit option takes
+        # precedence
+        schema = strawberry.Schema(Query)
+        assert SentryAsyncExtension in schema.extensions
+
+
+def test_sync_execution_uses_sync_extension(sentry_init):
+    sentry_init(integrations=[StrawberryIntegration(async_execution=False)])
+
+    with mock.patch(
+        "sentry_sdk.integrations.strawberry._get_installed_modules",
+        return_value={"fastapi": "0.103.1", "starlette": "0.27.0"},
+    ):
+        # actual installed modules should not matter, the explicit option takes
+        # precedence
+        schema = strawberry.Schema(Query)
+        assert SentrySyncExtension in schema.extensions
+
+
+def test_infer_execution_type_from_installed_packages_async(sentry_init):
+    sentry_init(integrations=[StrawberryIntegration()])
+
+    with mock.patch(
+        "sentry_sdk.integrations.strawberry._get_installed_modules",
+        return_value={"fastapi": "0.103.1", "starlette": "0.27.0"},
+    ):
+        schema = strawberry.Schema(Query)
+        assert SentryAsyncExtension in schema.extensions
+
+
+def test_infer_execution_type_from_installed_packages_sync(sentry_init):
+    sentry_init(integrations=[StrawberryIntegration()])
+
+    with mock.patch(
+        "sentry_sdk.integrations.strawberry._get_installed_modules",
+        return_value={"flask": "2.3.3"},
+    ):
+        schema = strawberry.Schema(Query)
+        assert SentrySyncExtension in schema.extensions
+
+
+def test_replace_existing_sentry_async_extension(sentry_init):
+    sentry_init(integrations=[StrawberryIntegration()])
+
+    schema = strawberry.Schema(Query, extensions=[SentryTracingExtension])
+    assert SentryTracingExtension not in schema.extensions
+    assert SentrySyncExtension not in schema.extensions
+    assert SentryAsyncExtension in schema.extensions
+
+
+def test_replace_existing_sentry_sync_extension(sentry_init):
+    sentry_init(integrations=[StrawberryIntegration()])
+
+    schema = strawberry.Schema(Query, extensions=[SentryTracingExtensionSync])
+    assert SentryTracingExtensionSync not in schema.extensions
+    assert SentryAsyncExtension not in schema.extensions
+    assert SentrySyncExtension in schema.extensions
+
+
+@pytest.mark.parametrize(
+    "client_factory,async_execution,framework_integrations",
+    (
+        (
+            "async_app_client_factory",
+            True,
+            [FastApiIntegration(), StarletteIntegration()],
+        ),
+        ("sync_app_client_factory", False, [FlaskIntegration()]),
+    ),
```

Comment:
It seems like you are using this same combination of parameters for most of the tests. So I would suggest that you somehow store these default parameters in a single variable, then reuse this everywhere. You could even I think define a custom decorator for all the tests like so:
```python
parameterize_strawberry_test = pytest.mark.parametrize(
    "client_factory,async_execution,framework_integrations",
    (
        (
            "async_app_client_factory",
            True,
            [FastApiIntegration(), StarletteIntegration()],
        ),
        ("sync_app_client_factory", False, [FlaskIntegration()]),
    ),
)
```
And then, you can decorate each of your tests with `@parameterize_strawberry_test`

Reply:
[Done!](https://github.com/getsentry/sentry-python/pull/2393/commits/ed52750d0e764e0755137ed243586a2faf420b11)